05_run_ocp.sh

```diff
@@ -106,7 +106,23 @@ ssh -o "StrictHostKeyChecking=no" core@$IP sudo modprobe iscsi_tcp
 # doesn't exist on the host
 ssh -o "StrictHostKeyChecking=no" core@$IP sudo mkdir /run/httpd
 
+# Internal dnsmasq should reserve IP addresses for each master
+cp -rf ironic/dnsmasq.conf ~stack/dnsmasq.conf
+for i in 0 1 2; do 
+  NODE_MAC=$(cat ~stack/ironic_nodes.json | jq -r ".nodes[${i}].ports[0].address")
+  NODE_IP="172.22.0.2${i}"
+  HOSTNAME="${CLUSTER_NAME}-etcd-${i}.${BASE_DOMAIN}"
+  # Make sure internal dnsmasq would assign an expected IP
+  echo "dhcp-host=${NODE_MAC},${HOSTNAME},${NODE_IP}" >> ~stack/dnsmasq.conf
+  # Reconfigure "external" dnsmasq
+  echo "${NODE_IP} ${HOSTNAME} ${CLUSTER_NAME}-api.${BASE_DOMAIN}" | sudo tee -a /etc/hosts.openshift
+  echo "srv-host=_etcd-server-ssl._tcp.${CLUSTER_NAME}.${BASE_DOMAIN},${HOSTNAME},2380,0,0" | sudo tee -a /etc/NetworkManager/dnsmasq.d/openshift.conf
+done
+sudo systemctl restart NetworkManager
+cat ~stack/dnsmasq.conf | ssh -o "StrictHostKeyChecking=no" core@$IP sudo dd of=dnsmasq.conf
```

Comment:
Does this mean that the dnsmasq that will be serving the deployment is from ironic? I would have expected ironic to only interact in the pxe booting and have the configuration be external. Is this something transitory?

Reply:
Yes, this is just for ironic containers to assign correct IP address. If libvirt xml can do that it should be the preferred way